### PR TITLE
dev-embedded/u-boot-tools: bump to 2020.10

### PR DESCRIPTION
**Pull request by jsmolic**

### CODE REVIEW

File: dev-embedded/u-boot-tools/u-boot-tools-2020.07.ebuild [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~arm64 ~ppc64 ~riscv ~x86"
+
+BDEPEND="
+	sys-devel/bison
+	sys-devel/flex"
+
+S=${WORKDIR}/${MY_P}
```
Comment (SoapGentoo) [inline dev-embedded/u-boot-tools/u-boot-tools-2020.07.ebuild:21]: quotes

**Review by SoapGentoo — CHANGES_REQUESTED** [review]

File: dev-embedded/u-boot-tools/u-boot-tools-2020.07.ebuild (hunk 1 above)
Comment (jsmolic) [reply dev-embedded/u-boot-tools/u-boot-tools-2020.07.ebuild]: Done ;)

File: dev-embedded/u-boot-tools/u-boot-tools-2020.10.ebuild [hunk 2]
```diff
@@ -12,59 +12,71 @@ SRC_URI="ftp://ftp.denx.de/pub/u-boot/${MY_P}.tar.bz2"
 
 LICENSE="GPL-2"
 SLOT="0"
-KEYWORDS="~amd64 ~arm ~arm64 ~ppc64 ~x86"
```
Comment (ajakk) [inline dev-embedded/u-boot-tools/u-boot-tools-2020.10.ebuild:15]: (Commenting on a different line because the review thing doesn't let me comment on a line you didn't change)

Why does the SRC_URI use the ftp protocol? It seems like the distfiles are also served over HTTPS, can you switch to that for security?

File: dev-embedded/u-boot-tools/u-boot-tools-2020.10.ebuild [hunk 3]
```diff
@@ -12,59 +12,71 @@ SRC_URI="ftp://ftp.denx.de/pub/u-boot/${MY_P}.tar.bz2"
...
-IUSE=""
+KEYWORDS="~amd64 ~arm ~arm64 ~ppc64 ~riscv ~x86"
+IUSE="minimal"
```
Comment (ajakk) [inline dev-embedded/u-boot-tools/u-boot-tools-2020.10.ebuild:16]: Why did you add this? It seems like the only difference is about 800kb on my system.

File: dev-embedded/u-boot-tools/u-boot-tools-2020.10.ebuild [hunk 4]
```diff
@@ -12,59 +12,71 @@ SRC_URI="ftp://ftp.denx.de/pub/u-boot/${MY_P}.tar.bz2"
...
-	dobin bmp_logo dumpimage fdtgrep gen_eth_addr img2srec mkenvimage mkimage
 	dobin env/fw_printenv
+
+	if ! use minimal; then
+		dobin bmp_logo dumpimage fdtgrep gen_eth_addr img2srec mkenvimage mkimage
+	fi
+
 	dosym fw_printenv /usr/bin/fw_setenv
 	insinto /etc
 	doins env/fw_env.config
```
Comment (ajakk) [inline dev-embedded/u-boot-tools/u-boot-tools-2020.10.ebuild:82]: Why not wrap the whole string passed to doman in quotes?

File: dev-embedded/u-boot-tools/u-boot-tools-2020.10.ebuild [hunk 5]
```diff
@@ -12,59 +12,71 @@ SRC_URI="ftp://ftp.denx.de/pub/u-boot/${MY_P}.tar.bz2"
...
+		tools-only_defconfig
+
+	if use minimal; then
+		_emake envtools
+	else
+		_emake tools-all
+	fi
 }
 
 src_test() { :; }
```
Comment (ajakk) [inline dev-embedded/u-boot-tools/u-boot-tools-2020.10.ebuild:70]: Whoops! Missed something in the other review. Why is src_test a no-op? It seems like there are tests, if they're broken they should be patched to run or they should have RESTRICT="test" (preferably with an explanation in comments).

File: dev-embedded/u-boot-tools/u-boot-tools-2020.10.ebuild [hunk 6]
```diff
@@ -12,59 +12,71 @@ SRC_URI="ftp://ftp.denx.de/pub/u-boot/${MY_P}.tar.bz2"
 
 LICENSE="GPL-2"
 SLOT="0"
-KEYWORDS="~amd64 ~arm ~arm64 ~ppc64 ~x86"
-IUSE=""
+KEYWORDS="~amd64 ~arm ~arm64 ~ppc64 ~riscv ~x86"
+IUSE="minimal"
 
 BDEPEND="
```
[the remark this answers is not in the document]
Comment (jsmolic) [reply dev-embedded/u-boot-tools/u-boot-tools-2020.10.ebuild:18]: Sure, this was mostly copy-paste bump, hence the style niggles

File: dev-embedded/u-boot-tools/u-boot-tools-2020.10.ebuild (hunk 5 above)
Comment (jsmolic) [reply dev-embedded/u-boot-tools/u-boot-tools-2020.10.ebuild:70]: Didn't go that far into it, it has been this way for several previous version... but I'll take a look

File: dev-embedded/u-boot-tools/u-boot-tools-2020.10.ebuild (hunk 6 above)
[the remark this answers is not in the document]
Comment (jsmolic) [reply dev-embedded/u-boot-tools/u-boot-tools-2020.10.ebuild:18]: Actually I prefer to do it this way if there are more deps, to keep the tabs consistent

File: dev-embedded/u-boot-tools/u-boot-tools-2020.10.ebuild (hunk 4 above)
Comment (jsmolic) [reply dev-embedded/u-boot-tools/u-boot-tools-2020.10.ebuild:82]: I prefer to quote only `${S}`, looks nicer

File: dev-embedded/u-boot-tools/u-boot-tools-2020.10.ebuild (hunk 3 above)
Comment (jsmolic) [reply dev-embedded/u-boot-tools/u-boot-tools-2020.10.ebuild:16]: Because I have a use-case where only envtools are needed, instead of all binaries, and that size can make a difference on embedded systems

**Comment by jsmolic** [comment]
This is done in https://github.com/gentoo/gentoo/commit/f4f8e821f7c750d3db6e6828a8cd70dc272c4dbd
